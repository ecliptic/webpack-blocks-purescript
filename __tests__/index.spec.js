// @flow
import purescript from '../index'

test('purescript()', () => {
  const block = purescript({template: 'assets/index.html'})
  const context = {fileType: () => /\.purs$/}
  const helpers = {
    merge: configSnippet => prevConfig => configSnippet,
  }

  block.pre(context, helpers)
  const result = block(context, helpers)({})

  expect(result).toMatchSnapshot()
})
