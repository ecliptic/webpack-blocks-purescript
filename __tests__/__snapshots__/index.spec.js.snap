// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`purescript() 1`] = `
Object {
  "module": Object {
    "rules": Array [
      Object {
        "exclude": /node_modules/,
        "loader": "purs-loader",
        "options": Object {
          "psc": "psa",
          "template": "assets/index.html",
          "warnings": false,
        },
        "test": /\\\\\\.purs\\$/,
      },
    ],
  },
  "resolve": Object {
    "extensions": Array [
      ".purs",
    ],
  },
}
`;
